iphone/cli/commands/_buildModule.js

```diff
@@ -592,82 +651,34 @@ iOSModuleBuilder.prototype.createUniversalBinary = function createUniversalBinar
 	}.bind(this));
 };
 
-iOSModuleBuilder.prototype.verifyBuildArch = function verifyBuildArch(next) {
+iOSModuleBuilder.prototype.verifyBuildArch = async function verifyBuildArch(next) {
 	this.logger.info(__('Verifying universal library'));
+	
+	const moduleId = this.isFramework ? this.moduleIdAsIdentifier  : this.moduleId;
+	const xcFrameworkPath = await path.join(this.projectDir, 'build', moduleId + '.xcframework');
+	const fileExists = await fs.existsSync(xcFrameworkPath);
+	if (!fileExists) {
+		this.logger.error(__(`Unable to find "${moduleId}.xcframework"`));
+		process.exit(1);
+	}
 
-	// TODO: We want to verify the set of targets/arches built
-	// The manifest shouldn't matter any more?
-	// Can we just ensure it's an xcframework?
-
-	const findLib = function (dest) {
-		const moduleId = this.isFramework ? this.moduleIdAsIdentifier  : this.moduleId;
-		const libName = this.isFramework ? this.moduleIdAsIdentifier + '.framework' : 'lib' + this.moduleId + '.a';
-		const lib = path.join(this.projectDir, 'build', moduleId + '.xcframework', dest, libName);
-		this.logger.info(__('Looking for ' + lib));
-
-		if (!fs.existsSync(lib)) {
-			return new Error(__('Unable to find the built %s library', 'Release-' + dest));
-		}
-		return lib;
-	}.bind(this);
-
-	const buildArchs = new Set();
+	const xcFrameworkInfo = await parsePlist(path.join(xcFrameworkPath, 'Info.plist'));
 
-	let lib = findLib('ios-arm64_armv7');
-	if (lib instanceof Error) {
-		// fallback to xcode 11 xcframework
-		lib = findLib('ios-armv7_arm64');
-		if (lib instanceof Error) {
-			this.logger.warn(__('The module is missing 64-bit support.'));
-		} else {
-			buildArchs.add('armv7');
-			buildArchs.add('arm64');
-		}
-	} else {
-		buildArchs.add('armv7');
-		buildArchs.add('arm64');
-	}
+	this.logger.info(__('Fetching available architectures in the module'));
 
-	lib = findLib('ios-arm64_i386_x86_64-simulator');
-	if (lib instanceof Error) {
-		// fall back to xcode 11 xcframework w/o arm64 sim support
-		lib = findLib('ios-i386_x86_64-simulator');
-		if (lib instanceof Error) {
-			// neither is available, so no sim support!
-			this.logger.warn(__('The module is missing ios simulator support.'));
+	for (const libInfo of xcFrameworkInfo.AvailableLibraries) {
+		const libraryPath = path.join(xcFrameworkPath, libInfo.LibraryPath);
```

Comment:
<!--
  0 failure: 
  1 warning:  iphone/cli/comman...
  
  
  DangerID: danger-id-Danger;
  File: iphone/cli/commands/_buildModule.js;
  Line: 670;
-->

- :warning: iphone/cli/commands/_buildModule.js line 670 – 'libraryPath' is assigned a value but never used. (no-unused-vars)